README.md

```diff
@@ -1,17 +1,137 @@
 
-## Workflow for new channels
+## Contents
 
-1) Head over to the [skeleton repo](https://github.com/laravel-notification-channels/skeleton) & press "Use this template" to create a repo from the skeleton.
-2) Use find/replace to replace all of the placeholders with the correct values (package name, author name, email, etc).
-3) Implement to logic for the channel & add tests.
-4) Fork this repo, add it as a remote and push your new channel to a branch.
-5) Submit a new PR against this repo for review.
+- [Installation](#installation)
+  - [Setting up the InterFAX service](#setting-up-the-InterFAX-service)
+- [Usage](#usage)
+  - [Available Message methods](#available-message-methods)
+- [Changelog](#changelog)
+- [Testing](#testing)
+- [Security](#security)
+- [Contributing](#contributing)
+- [Credits](#credits)
+- [License](#license)
 
-Take a look at our [FAQ](http://laravel-notification-channels.com/) to see our small list of rules, to provide top-notch notification channels.
+
+## Installation
+
+You can install this package via composer:
+
+```bash
+composer require laravel-notification-channels/interfax
+```
+
+The service provider gets loaded automatically.
+
+### Setting up the InterFAX service
+
+This channel will use your InterFAX username and password. To use the channel, add this to your `config/services.php` file:
+
+```php
+...
+'interfax' => [
+  'username' => env('INTERFAX_USERNAME'),
+  'password' => env('INTERFAX_PASSWORD'),
+  'pci'      => env('INTERFAX_PCI', false),
+],
+...
+```
+
+This will load your InterFAX credentials from the `.env` file. If your requests must be PCI-DSS-compliant, set `INTERFAX_PCI=true` in your `.env` file.
+
+## Usage
+
+To use this package, you can create a notification class, like `DocumentWasSent` from the example below, in your Laravel application. Make sure to check out [Laravel's documentation](https://laravel.com/docs/master/notifications) for this process.
+
+### Send PDF via fax
+
+```php
+<?php
+use NotificationChannels\Interfax\InterfaxChannel;
+use NotificationChannels\Interfax\InterfaxMessage;
+
+class DocumentWasSent extends Notification
+{
+
+  protected $files;
+
+  public function __construct(array $files)
+  {
+    $this->files = $files;
+  }
+
+    /**
+     * Get the notification's delivery channels.
+     *
+     * @param  mixed  $notifiable
+     * @return array
+     */
+    public function via($notifiable)
+    {
+        return [InterfaxChannel::class];
+    }
+
+    public function toInterfax($notifiable)
+    {
+        return (new InterfaxMessage)
+              ->files($this->files);
+    }
+}
+```
+
+The Notifiable model will need to return a destination fax number.
+
+```php
+public function routeNotificationForInterfax($notification)
+{
+    if($this->fax)
+        return preg_replace('/[^\d]/', '', $this->fax);
+
+    return null;
+}
+```
+
+### Available Message methods
+
+`file(string $file)` : Accepts the full path to a single file.  
```

Comment:
I think it would be worth documenting what types of files are accepted (I would assume pdf or image?)